packages/truffle-db/src/artifacts/schema.ts

```diff
@@ -75,5 +96,61 @@ export const schema = mergeSchemas({
       },
       
     },
+
+    Contract: {
+      name: {
+        fragment: "... on ContractObject { name: contractName }"
+      }, 
+      sourceContract: {
+        fragment: 
+        `... on ContractObject { 
+          ast
+        }`,
+        resolve: (obj, ast) => {
+          const { name, source } = obj;
+          const sourceContract = {
+            name: name, 
+            source: source, 
+            ast: ast
+          }
+          return sourceContract;
+        }
+      }, 
+    },
+
+    ContractConstructor: {
+      contract: {
+        fragment: 
+        `... on ContractObject { 
+          contents,
+          sourcePath,
+          ast
+        }`,
+        resolve: (obj, contents, sourcePath, ast) => {
```

Comment:
I think arguments are wrong here - I'd expect all the fields to be in `obj`, then on the next line we would do the destructuring.

Reply:
Yeah I had it as obj before, will change it back!